syntax="proto2";
package elasticfaiss;
option cc_generic_services = true;

import "shard.proto";

message CreateShardRequest {
    required string cluster = 1;
    required IndexConf conf = 2;
    required int32  idx = 3;
    required string shard_nodes = 4;
};

message CreateShardResponse {
    required bool success = 1;
    optional string error = 3;
};

message WorkNodeInitConf{
    repeated CreateShardRequest shards = 1;
}


message DeleteShardRequest {
    required string cluster = 1;
    required string name = 2;
    required int32  idx = 3;
};

message DeleteShardResponse {
    required bool success = 1;
    optional string error = 3;
};

service WorkNodeService {
    rpc create_shard(CreateShardRequest) returns (CreateShardResponse);
    rpc delete_shard(DeleteShardRequest) returns (DeleteShardResponse);
};


